cloud/ec2.go

```diff
@@ -367,6 +390,43 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
 	return nil
 }
 
+// setNextSubnet sets the subnet in the host's cached distro to the next one that supports this instance type.
+// If the current subnet doesn't support this instance type it's set to the first that does.
+func (m *ec2Manager) setNextSubnet(ctx context.Context, h *host.Host) error {
+	ec2Settings := &EC2ProviderSettings{}
+	if err := ec2Settings.FromDistroSettings(h.Distro, m.region); err != nil {
+		return errors.Wrap(err, "can't get provider settings")
+	}
+
+	supportingSubnets, err := typeCache.subnetsWithInstanceType(ctx, m.settings, m.client, h.InstanceType, ec2Settings.getRegion())
+	if err != nil {
+		return errors.Wrapf(err, "can't get supported subnets for instance type '%s'", h.InstanceType)
+	}
+	if len(supportingSubnets) == 0 {
+		return errors.Errorf("instance type '%s' is not supported by any configured subnet", h.InstanceType)
```

Comment:
Can this also include region for better error reporting?

Reply:
done